### PR TITLE
Refactoring Error Messages for the API

### PR DESCRIPTION
**Pull request by nsdeschenes**
As described.

### CODE REVIEW

File: api-js/src/affiliation/mutations/__tests__/update-user-role.test.js [hunk 1]
```diff
@@ -636,7 +632,7 @@ describe('update a users role', () => {
           )
 
           const error = [
-            new GraphQLError('Unable to update users role. Please try again.'),
+            new GraphQLError('Unable to update unknown users role.'),
```
Comment (Ethanljf) [inline api-js/src/affiliation/mutations/__tests__/update-user-role.test.js:635]: _user's_ 😉

File: api-js/src/affiliation/mutations/__tests__/update-user-role.test.js [hunk 2]
```diff
@@ -688,7 +684,9 @@ describe('update a users role', () => {
           )
 
           const error = [
-            new GraphQLError('Unable to update users role. Please try again.'),
+            new GraphQLError(
+              'Unable to update users role, in unknown organization.',
```
Comment (Ethanljf) [inline api-js/src/affiliation/mutations/__tests__/update-user-role.test.js:688]: See above!

File: api-js/src/affiliation/mutations/__tests__/update-user-role.test.js [hunk 3]
```diff
@@ -892,7 +894,7 @@ describe('update a users role', () => {
 
           const error = [
             new GraphQLError(
-              'Unable to update users role. Please invite user to the organization.',
+              'Unable to update users role that does not belong to this org. Please invite user to the organization.',
```
Comment (Ethanljf) [inline api-js/src/affiliation/mutations/__tests__/update-user-role.test.js:897]: -> "Unable to update role of user that does not belong to this org" ?

File: api-js/src/affiliation/mutations/update-user-role.js [hunk 4]
```diff
@@ -72,7 +70,7 @@ export const updateUserRole = new mutationWithClientMutationId({
       console.warn(
         `User: ${userKey} attempted to update a user: ${userName} role in org: ${orgId}, however there is no user associated with that user name.`,
       )
-      throw new Error(i18n._(t`Unable to update users role. Please try again.`))
+      throw new Error(i18n._(t`Unable to update unknown users role.`))
```
Comment (Ethanljf) [inline api-js/src/affiliation/mutations/update-user-role.js:73]: Same!

File: api-js/src/affiliation/mutations/update-user-role.js [hunk 5]
```diff
@@ -82,7 +80,9 @@ export const updateUserRole = new mutationWithClientMutationId({
       console.warn(
         `User: ${userKey} attempted to update a user: ${requestedUser._key} role in org: ${orgId}, however there is no org associated with that id.`,
       )
-      throw new Error(i18n._(t`Unable to update users role. Please try again.`))
+      throw new Error(
+        i18n._(t`Unable to update users role, in unknown organization.`),
```
Comment (Ethanljf) [inline api-js/src/affiliation/mutations/update-user-role.js:84]: Same!

File: api-js/src/affiliation/mutations/update-user-role.js [hunk 6]
```diff
@@ -116,7 +120,7 @@ export const updateUserRole = new mutationWithClientMutationId({
       )
       throw new Error(
         i18n._(
-          t`Unable to update users role. Please invite user to the organization.`,
+          t`Unable to update users role that does not belong to this org. Please invite user to the organization.`,
```
Comment (Ethanljf) [inline api-js/src/affiliation/mutations/update-user-role.js:123]: Same as other suggestion

File: api-js/src/affiliation/mutations/update-user-role.js [hunk 7]
```diff
@@ -150,7 +154,9 @@ export const updateUserRole = new mutationWithClientMutationId({
           `User: ${userKey} attempted to lower user: ${requestedUser._key} from ${affiliation.permission} to: admin.`,
         )
         throw new Error(
-          i18n._(t`Unable to update users role. Please try again.`),
+          i18n._(
+            t`Permission Denied: Please contact organization admin for help with updating users roles.`,
```
Comment (Ethanljf) [inline api-js/src/affiliation/mutations/update-user-role.js:158]: Same

File: api-js/src/affiliation/mutations/update-user-role.js [hunk 8]
```diff
@@ -169,7 +175,9 @@ export const updateUserRole = new mutationWithClientMutationId({
           `User: ${userKey} attempted to lower user: ${requestedUser._key} from ${affiliation.permission} to: user.`,
         )
         throw new Error(
-          i18n._(t`Unable to update users role. Please try again.`),
+          i18n._(
+            t`Permission Denied: Please contact organization admin for help with updating users roles.`,
```
Comment (Ethanljf) [inline api-js/src/affiliation/mutations/update-user-role.js:179]: Same

File: api-js/src/affiliation/mutations/update-user-role.js [hunk 9]
```diff
@@ -182,7 +190,11 @@ export const updateUserRole = new mutationWithClientMutationId({
       console.warn(
         `User: ${userKey} attempted to lower user: ${requestedUser._key} from ${affiliation.permission} to: ${role}.`,
       )
-      throw new Error(i18n._(t`Unable to update users role. Please try again.`))
+      throw new Error(
+        i18n._(
+          t`Permission Denied: Please contact organization admin for help with updating users roles.`,
```
Comment (Ethanljf) [inline api-js/src/affiliation/mutations/update-user-role.js:195]: Same

**Review by Ethanljf — COMMENTED** [review]
Just a few comments! Looks great otherwise!

File: api-js/src/affiliation/mutations/__tests__/update-user-role.test.js (hunk 1 above)
Comment (tparrott-cse) [reply api-js/src/affiliation/mutations/__tests__/update-user-role.test.js]: Unable to update role: user unknown.

File: api-js/src/affiliation/mutations/__tests__/update-user-role.test.js (hunk 3 above)
Comment (tparrott-cse) [reply api-js/src/affiliation/mutations/__tests__/update-user-role.test.js]: "Unable to update role of user that does not belong to this org"

This sounds.....weird?  "Unable to update role: user does not belong to organization"  Or something

File: api-js/src/affiliation/mutations/__tests__/update-user-role.test.js (hunk 2 above)
Comment (tparrott-cse) [reply api-js/src/affiliation/mutations/__tests__/update-user-role.test.js]: Unable to update role: organization unknown

File: api-js/src/affiliation/mutations/update-user-role.js (hunk 4 above)
Comment (tparrott-cse) [reply api-js/src/affiliation/mutations/update-user-role.js]: Unable to update role: user unknown

File: api-js/src/affiliation/mutations/update-user-role.js (hunk 8 above)
Comment (tparrott-cse) [reply api-js/src/affiliation/mutations/update-user-role.js]: This is fine

Comment (tparrott-cse) [reply api-js/src/affiliation/mutations/update-user-role.js]: The original change, that is

Comment (tparrott-cse) [reply api-js/src/affiliation/mutations/update-user-role.js]: Except it should say "for help with updating user roles"

Comment (tparrott-cse) [reply api-js/src/affiliation/mutations/update-user-role.js]: MOAR COMMENTS

**Review by Ethanljf — APPROVED** [review]

**Pull request merged**